### PR TITLE
#176 chore: add a more elegant way to create topics, don't raise errors in case a topic exists

### PR DESCRIPTION
**Pull request by zsinnema**
…rs in case a topic exists

### CODE REVIEW

File: libraries/cloudharness-common/cloudharness/events/client.py [hunk 1]
```diff
@@ -41,16 +41,29 @@ def create_topic(self):
...
+        for topic in [NewTopic(name=self.topic_id, num_partitions=1, replication_factor=1)]:
+            try:
+                # validate if the Topic already exists
+                admin_client.create_topics(new_topics=[topic], validate_only=True)
+                # if not then add it to the topic_list
+                topic_list.append(topic)
+            except TopicAlreadyExistsError as e:
+                pass
+            except Exception as e:
+                log.error(f"Ups... We had an error creating the new Topic --> {e}")
```
Comment (filippomc) [inline libraries/cloudharness-common/cloudharness/events/client.py:54]: log.error better add parameter exc_info=True to get the stack trace

File: libraries/cloudharness-common/cloudharness/events/client.py [hunk 2]
```diff
@@ -41,16 +41,29 @@ def create_topic(self):
...
             return admin_client.create_topics(new_topics=topic_list, validate_only=False)
         except TopicAlreadyExistsError as e:
-            log.error(f"Topic {self.topic_id} already exists.")
-            raise TopicAlreadyExistsError from e
+            # topic already exists "no worries", proceed
+            pass
         except Exception as e:
             log.error(f"Ups... We had an error creating the new Topic --> {e}")
             raise EventGeneralException from e
+        return True
```
Comment (filippomc) [inline libraries/cloudharness-common/cloudharness/events/client.py:66]: What the return value is meant for?

File: libraries/cloudharness-common/cloudharness/events/client.py [hunk 3]
```diff
@@ -41,16 +41,29 @@ def create_topic(self):
                                         client_id=self.client_id)
         # ## Create topic
 
-        topic_list = [NewTopic(name=self.topic_id, num_partitions=1, replication_factor=1)]
+        topic_list = []
+        for topic in [NewTopic(name=self.topic_id, num_partitions=1, replication_factor=1)]:
```
Comment (filippomc) [inline libraries/cloudharness-common/cloudharness/events/client.py:45]: Why use this fake list with one topic?

**Review by filippomc — COMMENTED** [review]
Added a suggestion and a few questions

File: libraries/cloudharness-common/cloudharness/events/client.py (hunk 3 above)
Comment (zsinnema) [reply libraries/cloudharness-common/cloudharness/events/client.py:45]: kafka client create_topics supports creation of multiple topics at once

File: libraries/cloudharness-common/cloudharness/events/client.py (hunk 2 above)
Comment (zsinnema) [reply libraries/cloudharness-common/cloudharness/events/client.py]: agree, removed

File: libraries/cloudharness-common/cloudharness/events/client.py (hunk 1 above)
Comment (zsinnema) [reply libraries/cloudharness-common/cloudharness/events/client.py]: done and updated also the other log error messages

File: libraries/cloudharness-common/cloudharness/events/client.py (hunk 3 above)
Comment (zsinnema) [reply libraries/cloudharness-common/cloudharness/events/client.py]: refactored it

**Review by filippomc — APPROVED** [review]
lgtm

**Pull request merged**